source/plugin/data/index.rst

```diff
@@ -0,0 +1,78 @@
+oblivious to what data it holds. While this may appear less straightforward than direct accessor methods, it is
+foremost far more extensible. And thanks to the addition of ``Key``\ s, simply accessing specific values is no
+less straightforward.
+
+.. warning::
+
+    As of writing, the Data API is not yet fully implemented. Refer to the `Implementation Tracker
+    <https://github.com/SpongePowered/SpongeCommon/issues/8>`_, ask in the ``#spongedev`` IRC channel or on the
+    `Forums <https://forums.spongepowered.org>`_ to find out if the data you need to work with is available yet.
+
+Concepts
+========
+
+On first glance at the API docs, the data API threatens to overwhelm you with lots of interfaces and packages. But
+to simply use the data API, you will not have to deal with many of them, as most interfaces found there are just
+specific data manipulators.
+
+DataHolder
+~~~~~~~~~~
+
+A data holder is just that - something that holds data. It provides methods to retrieve and offer back data. The
+interface itself is completely oblivious to the type of data held. Since only the implementations will possess
+this knowledge, it is possible to ask a ``DataHolder`` to provide data it does not have or to accept data it
+cannot use. In those cases, the return values of the methods will provide the information that data is not
+available (via ``Optional.absent()``) or not accepted.
+
+Property
+~~~~~~~~
+
+A property is a read-only point of data. It provides information that is directly derived from its holder's type,
+but, unlike other data, it is unchangeable (except for core mods). Examples of this are the harvesting ablities on tools
+(represented as ``HarvestingProperty``) or the damage absorption of an equippable armor item.
+
+DataManipulator
+~~~~~~~~~~~~~~~
+
+A data manipulator represents points of cohesive data that describes a certain component of its holder. For
+example ``HealthData``, which contains both current and maximum health. If a data holder has ``HealthData``, it
+has health that can somehow be depleted and replenished and can die if that health is depleted. This allows for
+the re-use of such components over the API and prevents duplication of accessor methods. For example, sheep, stained
+glass blocks and leather armor all can share the ``DyeableData`` holding the color they are dyed in.
+
+Key
+~~~
+
+A ``Key`` is a unique identifier for a single point of data and can be used to directly read or set that point of
+data without worrying about data manipulators. It was designed to provide a convenient way of accessing data
+similar to direct getter/setter methods. All keys used within Sponge are listed as constants in the
+``org.spongepowered.api.data.key.Keys`` utility class.
+
+Value
+~~~~~
+
+Within the Data API, a value referred to by a ``Key`` is encoded in a container object. For this documentation, it
+is referred to as 'value container' to avoid confusion with the actual value, but note that the class name
```

Comment:
Should likely find a different word to really avoid confusion as to which `ValueContainer` you're referring to. Maybe a "keyed value" to represent a `Value`.